# Prepare data

## Bioinformatics

### Get data

```{sh get_genome, warning=FALSE, comments="", message=FALSE, eval=FALSE}
mkdir genome
wget -P genome https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/009/045/GCF_000009045.1_ASM904v1/GCF_000009045.1_ASM904v1_genomic.fna.gz
```

```{sh get_reads, warning=FALSE, comments="", message=FALSE, eval=FALSE}
mkdir reads
# fetch data from ERDA
```

### Run annotation and profiling

```{sh run_annotation, warning=FALSE, comments="", message=FALSE, eval=FALSE}
drakkar annotating -b genome
drakkar expressing -b genome -r reads
```

### Get working files

- gene_annotations.tsv.xz
- gene_counts.tsv.xz

## Load data

```{r load_reads, warning=FALSE, comments="", message=FALSE}
read_counts <- read_tsv(
  "data/gene_counts.tsv.xz",
  comment = "#",          # skip featureCounts metadata
  col_names = TRUE,       # header is present
  show_col_types = FALSE  # silence messages
) %>%
  rename(gene = Geneid) %>% 
  pivot_longer(!c(gene,Chr,Start,End,Strand,Length),names_to="library",values_to = "counts") %>% 
  mutate(sample = str_extract(library, "^[^_]+")) %>% 
  group_by(gene,Chr,Start,End,Strand,Length,sample) %>% 
  summarise(counts=sum(counts)) %>% 
  pivot_wider(c(gene,Chr,Start,End,Strand,Length), names_from = "sample", values_from = "counts") 

```

```{r load_annotations, warning=FALSE, comments="", message=FALSE}
gene_annotations <- read_tsv("data/gene_annotations.tsv.xz") %>%
    rename(gene=1)
```

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_metadata <- read_csv("data/sample_metadata.csv") 
```

## Preliminary visualisations

```{r seqdepth_overview, warning=FALSE, comments="", message=FALSE, eval=FALSE}
read_counts %>% 
  pivot_longer(!c(gene,Chr,Start,End,Strand,Length),names_to="sample",values_to = "counts") %>% 
  group_by(sample) %>% 
  summarise(counts=sum(counts)) %>% 
  ggplot(aes(x=sample,y=counts)) +
    geom_bar(stat="identity")
```

```{r kegg_overview, warning=FALSE, comments="", message=FALSE}
inner_join(read_counts,gene_annotations %>% select(gene,kegg),by="gene") %>% 
  pivot_longer(!c(gene,Chr,Start,End,Strand,Length,kegg),names_to="sample",values_to = "counts") %>% 
  group_by(sample,kegg) %>% 
  summarise(counts=sum(counts)) %>% 
  ggplot(aes(x=sample,y=kegg,fill=log(counts))) +
    geom_tile()
```

Dwfssdf semgf sdgm